### PR TITLE
Fix function pointers/delegates on Harvard architectures

### PR DESCRIPTION
**Pull request by kinke**
Resolves issue #4432 by setting the address space of such IR pointers.

### CODE REVIEW

File: tests/compilable/funcptr_harvard_gh4432.d [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{
+    fp();
+    dg();
+}
+
+void bar()
+{
+    foo(() {}, delegate() {});
+
+    FP fp = &bar;
```
Comment (JohanEngelen) [inline tests/compilable/funcptr_harvard_gh4432.d:20]: Probably good to also add testing of (normal) pointers to variables.
No IR test to validate that the address-space is correct? Or do we assume that is already part of LLVM checking?

Comment (kinke) [reply tests/compilable/funcptr_harvard_gh4432.d]: Yeah I was a bit lazy and wasn't sure how far we'd get with this. I've extended the test now, incl. storing and loading a function pointer in static *data*.

**Comment by dukc** [comment]
Great, thanks! I suggest a bit more stress-testing though: assigning a function pointer to a void pointer, pointer arithmetic and casting to/from an integral.

**Comment by kinke** [comment]
> I suggest a bit more stress-testing though: assigning a function pointer to a void pointer, pointer arithmetic and casting to/from an integral.

Heh right, that's exactly what I don't expect to work yet ('not sure how far we'd get with this').

**Comment by dukc** [comment]
> Heh right, that's exactly what I don't expect to work yet ('not sure how far we'd get with this').

Understood :smile: . Function pointers/delegates aren't quite fixed though in that case. Granted, the original issue still is.

**Pull request merged**